crates/topos-tce-broadcast/src/sampler/mod.rs

```diff
@@ -355,86 +374,62 @@ impl Sampler {
     fn reset_echo_subscription_sample(&mut self) {
         self.pending_subscriptions.echo.clear();
 
-        let echo_sizer = |len| min(len, self.params.echo_sample_size);
-        match sample_reduce_from(&self.visible_peers, echo_sizer) {
-            Ok(echo_candidates) => {
-                debug!(
-                    "Start the reset of the the Echo Sample currently composed by: {:?}",
-                    echo_candidates
-                );
-
-                for peer in &echo_candidates.value {
-                    info!("Adding the Peer {peer} to the pending Echo Subscriptions");
-                    self.pending_subscriptions.echo.insert(*peer);
-                }
+        debug!(
+            "Start the reset of the the Echo Sample currently composed by: {:?}",
+            &self.visible_peers
+        );
 
-                if let Err(error) = self.event_sender.send(ProtocolEvents::EchoSubscribeReq {
-                    peers: echo_candidates.value,
-                }) {
-                    error!("Unable to send event {:?}", error);
-                }
-            }
-            Err(e) => {
-                error!("Failed to create the sample for the Echo Subscriptions: {e:?}");
-            }
+        for peer in &self.visible_peers {
+            info!("Adding the Peer {peer} to the pending Echo Subscriptions");
+            self.pending_subscriptions.echo.insert(*peer);
+        }
```

Comment:
```suggestion
     self.pending_subscriptions.echo = self.visible_peers.iter().cloned().collect();
```

Reply:
Also on other similar places.

Reply:
All the peers from the visible peers are already printed above with debug? `info` is too verbose I think